src/backend/access/external/pxfheaders.c

```diff
@@ -158,6 +166,29 @@ static void add_tuple_desc_httpheader(CHURL_HEADERS headers, Relation rel)
 	pfree(formatter.data);
 }
 
+static void add_projection_desc_httpheader(CHURL_HEADERS headers, ProjectionInfo *projInfo) {
+	int i;
+	char long_number[sizeof(int32) * 8];
+	int *varNumbers = projInfo->pi_varNumbers;
+	StringInfoData formatter;
+	initStringInfo(&formatter);
+
+    /* Convert the number of projection columns to a string */
+    pg_ltoa(list_length(projInfo->pi_targetlist), long_number);
+    churl_headers_append(headers, "X-GP-ATTRS-PROJ", long_number);
+
+	for(i = 0; i < list_length(projInfo->pi_targetlist); i++) {
```

Comment:
Yes it will be in another PR related to [this](https://issues.apache.org/jira/browse/HAWQ-583?jql=project%20%3D%20HAWQ%20AND%20resolution%20%3D%20Unresolved%20AND%20assignee%20%3D%20kavinderd%20ORDER%20BY%20priority%20DESC) Jira.